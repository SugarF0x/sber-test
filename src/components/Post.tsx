import FavoriteIcon from "@material-ui/icons/Favorite";
import React        from "react";

import { connect }     from "react-redux";
import { setFavorite } from "../store/actions";

import {
  Card,
  CardActions,
  CardContent,
  CardHeader,
  IconButton,
} from "@material-ui/core";

import { IPost } from "../store/types";

interface IPostProps {
  post: IPost;
  favorite: boolean;
  setFavorite: Function;
}

interface IPostState {
  fav: boolean;
}

export class Post extends React.Component<IPostProps, IPostState> {
  card           = {
    margin: '1rem',
  };
  favoriteButton = {
    display:        'flex',
    justifyContent: 'space-between',
  };

  // util functions

  trimContent = (str: string): string => {
    let start = str.indexOf('<p>');
    let end   = str.indexOf('</p>');
    if (str.slice(start + 3, end).indexOf('<') !== -1) {
      return this.trimContent(str.slice(end + 3));
    } else {
      return str.slice(start + 3, end);
    }
  }

  formatDate(str: string): string {
    let date = new Date(str);

    let day   = date.getDate().toString().length === 1 ? `0${ date.getDate() }` : date.getDate();
    let month = date.getMonth().toString().length === 1 ? `0${ date.getMonth()+1 }` : date.getMonth()+1;
    let year  = date.getFullYear();
    let hour  = date.getHours().toString().length === 1 ? `0${ date.getHours() }` : date.getHours();
    let mins  = date.getMinutes().toString().length === 1 ? `0${ date.getMinutes() }` : date.getMinutes();

    return `${ day }.${ month }.${ year } at ${ hour }:${ mins }`
  }

  favoriteAction = () => {
    this.props.setFavorite(this.props.post)
  }

  render() {
    return (
      <Card style={ this.card } variant="outlined">
        <CardHeader title={ this.props.post.title }/>
        <CardContent>
          { this.trimContent(this.props.post.description) }
        </CardContent>
        <CardActions style={ this.favoriteButton }>
          <span>{ this.formatDate(this.props.post.created_at) }</span>
          <IconButton aria-label="add to favorites" onClick={ this.favoriteAction }>
            <FavoriteIcon color={ this.props.favorite ? 'secondary' : 'action' }/>
          </IconButton>
        </CardActions>
      </Card>
    );
  }
}

const mapDispatchToProps = {
  setFavorite,
};

export default connect(null, mapDispatchToProps)(Post);
