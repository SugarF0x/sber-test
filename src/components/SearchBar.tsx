import React from 'react';

import { connect }                   from "react-redux";
import { toggleDisplay, fetchPosts } from "../store/actions";

import { Input, Button }          from '@material-ui/core';
import { IRootPosts, IRootState } from "../store/types";

interface ISearchProps {
  posts: IRootPosts;
  toggleDisplay: Function;
  fetchPosts: Function;
}

interface ISearchState {
  description: string,
  location: string,
}

export class SearchBar extends React.Component<ISearchProps, ISearchState> {
  constructor(props: ISearchProps) {
    super(props);
    this.state = {
      description: '',
      location:    '',
    };
  }

  style = {
    wrap: {
      display:  'flex',
      flexFlow: 'column',
    },
  };

  joinFilter = (): string => {
    return `${ this.state.description }-${ this.state.location }`
  }

  searchHandler = () => {
    if (this.joinFilter() === this.props.posts.filter) {
      this.props.toggleDisplay();
    } else {
      this.props.fetchPosts(this.state.description, this.state.location);
    }
  }

  displayFavorites = () => {
    this.props.toggleDisplay();
  }

  enterHandler = (e: any) => {
    if (e.key === 'Enter' && (this.state.location || this.state.description)) {
      this.searchHandler();
    }
  }

  render() {
    return (
      <div>
        <div style={ this.style.wrap }>
          <Input placeholder="Description"
                 value={ this.state.description }
                 onChange={ (e) => {
                   this.setState({description: e.target.value});
                 } }
                 onKeyDown={ e => this.enterHandler(e) }
          />
          <Input placeholder="Location"
                 value={ this.state.location }
                 onChange={ (e) => {
                   this.setState({location: e.target.value});
                 } }
                 onKeyDown={ e => this.enterHandler(e) }
          />
        </div>
        <Button style={ {margin: '1rem'} }
                variant="contained"
                color="primary"
                onClick={ () => {
                  /**
                   * these functions can't be fixed as typsecript considers them unasignable
                   * this will not be an issue with regular js
                   */
                  this.searchHandler();
                } }
                disabled={ !(this.state.location || this.state.description)
                           || this.props.posts.status === 'fetching'
                           || ((this.joinFilter() === this.props.posts.filter)
                                && (this.props.posts.display !== 'favs')) }
        >
          Get data
        </Button>
        <Button style={ {margin: '1rem'} }
                variant="contained"
                color="primary"
                onClick={ () => {
                  /**
                   * these functions can't be fixed as typsecript considers them unasignable
                   * this will not be an issue with regular js
                   */
                  this.displayFavorites();
                } }
                disabled={ this.props.posts.favs.length === 0
                           || this.props.posts.display === 'favs'
                           || this.props.posts.status === 'fetching' }
        >
          Get favorites
        </Button>
      </div>
    );
  }
}

const mapStateToProps    = (state: IRootState) => ({
  posts: state.posts
});
const mapDispatchToProps = (dispatch: Function) => {
  return {
    toggleDisplay: () => dispatch(toggleDisplay()),
    fetchPosts:    (desc: string, loc: string) => dispatch(fetchPosts(desc, loc))
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
