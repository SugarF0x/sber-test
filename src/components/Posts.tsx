import React from 'react';

import { Button, CircularProgress } from "@material-ui/core";

import Post      from './Post';
import SearchBar from './SearchBar';

import { connect }       from "react-redux";
import { getDummyPosts } from "../store/actions";

import { IPost, IRootState, IRootPosts } from '../store/types';

interface IWrapProps {
  posts: IRootPosts;
  getDummyPosts: Function;
}

interface IWrapState {

}

class Wrapper extends React.Component<IWrapProps, IWrapState> {
  pickPosts = () => {
    if (this.props.posts.display === 'default') {
      return (
        <div>
          <h3>There are currently no posts here</h3>
          <Button style={ { margin: '1rem' } }
                  variant="contained"
                  color="primary"
                  onClick={ () => {
                    /**
                     * these functions can't be fixed as typsecript considers them unasignable
                     * this will not be an issue with regular js
                     */
                    this.props.getDummyPosts()
                  } }
          >
            Get dummy data
          </Button>
        </div>
      )
    } else {
      switch (this.props.posts.status) {
        case 'success':
          return this.props.posts[this.props.posts.display].map((entry: IPost) => (
            <Post key={ entry.id } post={ entry } favorite={ this.props.posts.favs.filter((e: IPost) => entry.id === e.id).length > 0 }/>
          ))
        case 'fetching':
          return (
            <div>
              <h3>Fetching data...</h3>
              <CircularProgress/>
            </div>
          )
        case 'not_found':
          return (
            <h3>No search results found for this query</h3>
          )
        default:
          return (
            <h3>Oops!.. Something went wrong!</h3>
          )
      }
    }
  }

  render() {
    return (
      <div>
        <h1>Posts from GitHub</h1>
        <SearchBar/>
        { this.pickPosts() }
      </div>
    );
  }
}

const mapStateToProps    = (state: IRootState) => ({
  posts: state.posts
});
const mapDispatchToProps = {
  getDummyPosts,
};

export default connect(mapStateToProps, mapDispatchToProps)(Wrapper);
