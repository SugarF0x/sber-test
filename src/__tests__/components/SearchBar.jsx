import React         from "react";
import { mount }     from "enzyme";
import { SearchBar } from "../../components/SearchBar";

describe("Search Component Testing", () => {
  const store     = {
    status:  "success",
    search:  [],
    favs:    [],
    filter:  "",
    display: "default",
  };
  const dummyPost = {
    id:           "569bfcdc-fad1-4563-ae57-8585831db596",
    type:         "Full Time",
    url:          "https://jobs.github.com/positions/569bfcdc-fad1-4563-ae57-8585831db596",
    created_at:   "Sun Sep 06 14:30:27 UTC 2020",
    company:      "Hotel Trades Council, AFL-CIO",
    company_url:  "http://www.hotelworkers.org",
    location:     "New York, NY",
    title:        "Web Developer",
    description:  "<p>The New York Hotel Trades Council, AFL-CIO, a labor union that represents nearly 40,000 workers in the hotel, gaming, and racing industries in the New York City area, seeks an experienced web developer.</p>\n<p>This COVID crisis has shown us the immense value to our Union of developing a more expansive and sophisticated website. Since March, the number of viewers to our site has increased seven-fold (to as many as 20,000 per day and more) and we are investing our resources accordingly. In fact, we have so many concrete ideas to put the site to new innovative uses, that our one full-time Web Developer has much more work to do than he can possibly handle. Among the new uses we are developing are:</p>\n<p>Conducting secret ballot ratification votes online;\nProviding multi-lingual content to an incredibly diverse membership (the current site supports English, Spanish and Chinese content);\nMobilizing rank-and-file political action in a private individually personalized Member Section;\nProviding non-union workers with a confidential communication channel with the union and interactive pages allowing them to compare their wages, benefits and working conditions to those enjoyed by unionized workers; and\nMany other applications.</p>\n<p>Our Union constantly seeks new ways to increase our effectiveness through smart use of technology and we are looking for a creative developer to help conceive of, design and implement tools to support the Union in our mission.</p>\n<p>This position offers the prospect of a great career and, as we are a labor union, an excellent benefit package.</p>\n<p>The team currently works remotely from home due to the COVID situation. When things return to normal, we intend to be flexible about remote work, and will consider skilled applicants who don’t live in the New York area.</p>\n<p>Starting salary: $70,000 to $85,000</p>\n<p>Benefits include: Medical, Dental, Optical, and Prescription Drugs with family coverage (including domestic partners); Pension, Pre-paid Legal, Flexible Spending Accounts, Life Cycle Benefits, Long-Term Care Insurance, and other benefits.</p>\n<p>Requirements:\nAt least 3 years developing websites using an MVC framework (with preference given to specific knowledge of Laravel PHP framework).\nStrong UI/UX design skills\nStrong javascript skills</p>\n<p>Other useful experience/qualifications:</p>\n<ul>\n<li>Advanced experience managing SQL databases</li>\n<li>Linux server administration</li>\n<li>Networking skills</li>\n<li>Python</li>\n</ul>\n",
    how_to_apply: "<p>Please submit a resume, three references and a cover letter to <a href=\"mailto:careers@nyhtc.org\">careers@nyhtc.org</a> with the subject line “Web Developer.”  The cover letter should be clear and concise and explain the specific reasons for your interest in the position, and specify (with appropriate explanation) which of each of the listed qualifications you possess and which ones you do not possess.</p>\n<p>The New York Hotel Trades Council, AFL-CIO, is an equal opportunity employer.  All persons regardless of age, race, ethnicity, gender, religious affiliation and sexual orientation are encouraged to apply.</p>\n",
    company_logo: "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbE9KIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--a30d608a910769a14438f7e0d7f3a673ca961499/HTC%20logo.jpg",
  };
  let wrapper;

  /**
   * override wrapper props with toMutate keeping the rest intact
   */
  function mutateProps(toMutate) {
    wrapper.setProps({ posts: Object.assign({}, wrapper.props().posts, toMutate) });
  }

  beforeEach(() => {
    // noinspection RequiredAttributes
    wrapper = mount(<SearchBar posts={ store }/>);
  });

  describe("Element render tests", () => {
    test("Render component", () => {
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });
    test("Render 2 text fields", () => {
      expect(wrapper.find("input")).toHaveLength(2);
    });
    test("Render 2 buttons", () => {
      expect(wrapper.find("button")).toHaveLength(2);
    });
  });

  describe("Element interaction tests", () => {
    test("Search button disabled state change on input", () => {
      let description = wrapper.find("input").at(0);
      let location    = wrapper.find("input").at(1);
      /*
       i wanted to do a button variable
       but for some reason it's state would not update on change
       while the full path call works every time
       ?question mark?
       */
      /**
       * Initial state test
       */
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(true);

      /**
       * Stage change on Description input
       */
      description.simulate("change", { target: { value: "js" } });
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(false);
      description.simulate("change", { target: { value: "" } });

      /**
       * State change on Location input
       */
      location.simulate("change", { target: { value: "moscow" } });
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(false);
      location.simulate("change", { target: { value: "" } });

      /**
       * State change on both Description and Location input
       */
      description.simulate("change", { target: { value: "js" } });
      location.simulate("change", { target: { value: "moscow" } });
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(false);
      description.simulate("change", { target: { value: "" } });
      location.simulate("change", { target: { value: "" } });

      /**
       * State reversion to initial on input clear
       */
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(true);
    });
    test("Search button availability based on filter", () => {
      /**
       * Initial test
       */
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(true);

      /**
       * Set matching inputs and filter
       * (mismatching is already tested before)
       */
      wrapper.setProps({ posts: Object.assign({}, store, { filter: "js-moscow" }) });
      mutateProps({ filter: "js-moscow" });
      wrapper.find("input").at(0).simulate("change", { target: { value: "js" } });
      wrapper.find("input").at(1).simulate("change", { target: { value: "moscow" } });
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(true);

      /**
       * Change inputs
       */
      wrapper.find("input").at(0).simulate("change", { target: { value: "python" } });
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(false);
    });
    test("Favorites' button disabled state change on favs content change", () => {
      /**
       * Initial state test
       */
      expect(wrapper.find("button").at(1).prop("disabled")).toBe(true);

      /**
       * Add 2 favorites
       */
      mutateProps({ favs: [dummyPost, dummyPost] });
      expect(wrapper.find("button").at(1).prop("disabled")).toBe(false);

      /**
       * Remove 1 favorite
       */
      mutateProps({ favs: [dummyPost] });
      expect(wrapper.find("button").at(1).prop("disabled")).toBe(false);

      /**
       * Remove all favorites
       */
      mutateProps({ favs: [] });
      expect(wrapper.find("button").at(1).prop("disabled")).toBe(true);
    });
    test("Buttons' state changed based on display", () => {
      wrapper.setProps({
        posts: Object.assign({}, store, {
          filter: "",
          search: [dummyPost],
          favs:   [dummyPost],
        }),
      });
      wrapper.find("input").at(0).simulate("change", { target: { value: "js" } });

      /**
       * Initial test with both input and favorites present
       */
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(false);
      expect(wrapper.find("button").at(1).prop("disabled")).toBe(false);

      /**
       * Change display to 'favs'
       */
      mutateProps({ display: "favs" });
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(false);
      expect(wrapper.find("button").at(1).prop("disabled")).toBe(true);

      /**
       * Change display to 'search'
       */
      mutateProps({ display: "search", filter: "js-" });
      expect(wrapper.find("button").at(0).prop("disabled")).toBe(true);
      expect(wrapper.find("button").at(1).prop("disabled")).toBe(false);
    });
  });
});
