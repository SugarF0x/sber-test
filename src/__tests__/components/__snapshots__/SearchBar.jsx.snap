// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Testing Element render tests Render component 1`] = `
<SearchBar
  posts={
    Object {
      "display": "default",
      "favs": Array [],
      "filter": "",
      "search": Array [],
      "status": "success",
    }
  }
>
  <div>
    <div
      style={
        Object {
          "display": "flex",
          "flexFlow": "column",
        }
      }
    >
      <WithStyles(ForwardRef(Input))
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Description"
        value=""
      >
        <ForwardRef(Input)
          classes={
            Object {
              "colorSecondary": "MuiInput-colorSecondary",
              "disabled": "Mui-disabled",
              "error": "Mui-error",
              "focused": "Mui-focused",
              "formControl": "MuiInput-formControl",
              "fullWidth": "MuiInput-fullWidth",
              "input": "MuiInput-input",
              "inputMarginDense": "MuiInput-inputMarginDense",
              "inputMultiline": "MuiInput-inputMultiline",
              "inputTypeSearch": "MuiInput-inputTypeSearch",
              "marginDense": "MuiInput-marginDense",
              "multiline": "MuiInput-multiline",
              "root": "MuiInput-root",
              "underline": "MuiInput-underline",
            }
          }
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Description"
          value=""
        >
          <WithStyles(ForwardRef(InputBase))
            classes={
              Object {
                "colorSecondary": "MuiInput-colorSecondary",
                "disabled": "Mui-disabled",
                "error": "Mui-error",
                "focused": "Mui-focused",
                "formControl": "MuiInput-formControl",
                "fullWidth": "MuiInput-fullWidth",
                "input": "MuiInput-input",
                "inputMarginDense": "MuiInput-inputMarginDense",
                "inputMultiline": "MuiInput-inputMultiline",
                "inputTypeSearch": "MuiInput-inputTypeSearch",
                "marginDense": "MuiInput-marginDense",
                "multiline": "MuiInput-multiline",
                "root": "MuiInput-root MuiInput-underline",
                "underline": null,
              }
            }
            fullWidth={false}
            inputComponent="input"
            multiline={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Description"
            type="text"
            value=""
          >
            <ForwardRef(InputBase)
              classes={
                Object {
                  "adornedEnd": "MuiInputBase-adornedEnd",
                  "adornedStart": "MuiInputBase-adornedStart",
                  "colorSecondary": "MuiInputBase-colorSecondary MuiInput-colorSecondary",
                  "disabled": "Mui-disabled Mui-disabled",
                  "error": "Mui-error Mui-error",
                  "focused": "Mui-focused Mui-focused",
                  "formControl": "MuiInputBase-formControl MuiInput-formControl",
                  "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                  "input": "MuiInputBase-input MuiInput-input",
                  "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                  "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                  "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                  "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                  "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                  "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                  "marginDense": "MuiInputBase-marginDense MuiInput-marginDense",
                  "multiline": "MuiInputBase-multiline MuiInput-multiline",
                  "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                }
              }
              fullWidth={false}
              inputComponent="input"
              multiline={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Description"
              type="text"
              value=""
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <input
                  className="MuiInputBase-input MuiInput-input"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Description"
                  type="text"
                  value=""
                />
              </div>
            </ForwardRef(InputBase)>
          </WithStyles(ForwardRef(InputBase))>
        </ForwardRef(Input)>
      </WithStyles(ForwardRef(Input))>
      <WithStyles(ForwardRef(Input))
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Location"
        value=""
      >
        <ForwardRef(Input)
          classes={
            Object {
              "colorSecondary": "MuiInput-colorSecondary",
              "disabled": "Mui-disabled",
              "error": "Mui-error",
              "focused": "Mui-focused",
              "formControl": "MuiInput-formControl",
              "fullWidth": "MuiInput-fullWidth",
              "input": "MuiInput-input",
              "inputMarginDense": "MuiInput-inputMarginDense",
              "inputMultiline": "MuiInput-inputMultiline",
              "inputTypeSearch": "MuiInput-inputTypeSearch",
              "marginDense": "MuiInput-marginDense",
              "multiline": "MuiInput-multiline",
              "root": "MuiInput-root",
              "underline": "MuiInput-underline",
            }
          }
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Location"
          value=""
        >
          <WithStyles(ForwardRef(InputBase))
            classes={
              Object {
                "colorSecondary": "MuiInput-colorSecondary",
                "disabled": "Mui-disabled",
                "error": "Mui-error",
                "focused": "Mui-focused",
                "formControl": "MuiInput-formControl",
                "fullWidth": "MuiInput-fullWidth",
                "input": "MuiInput-input",
                "inputMarginDense": "MuiInput-inputMarginDense",
                "inputMultiline": "MuiInput-inputMultiline",
                "inputTypeSearch": "MuiInput-inputTypeSearch",
                "marginDense": "MuiInput-marginDense",
                "multiline": "MuiInput-multiline",
                "root": "MuiInput-root MuiInput-underline",
                "underline": null,
              }
            }
            fullWidth={false}
            inputComponent="input"
            multiline={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Location"
            type="text"
            value=""
          >
            <ForwardRef(InputBase)
              classes={
                Object {
                  "adornedEnd": "MuiInputBase-adornedEnd",
                  "adornedStart": "MuiInputBase-adornedStart",
                  "colorSecondary": "MuiInputBase-colorSecondary MuiInput-colorSecondary",
                  "disabled": "Mui-disabled Mui-disabled",
                  "error": "Mui-error Mui-error",
                  "focused": "Mui-focused Mui-focused",
                  "formControl": "MuiInputBase-formControl MuiInput-formControl",
                  "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                  "input": "MuiInputBase-input MuiInput-input",
                  "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                  "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                  "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                  "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                  "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                  "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                  "marginDense": "MuiInputBase-marginDense MuiInput-marginDense",
                  "multiline": "MuiInputBase-multiline MuiInput-multiline",
                  "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                }
              }
              fullWidth={false}
              inputComponent="input"
              multiline={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Location"
              type="text"
              value=""
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <input
                  className="MuiInputBase-input MuiInput-input"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Location"
                  type="text"
                  value=""
                />
              </div>
            </ForwardRef(InputBase)>
          </WithStyles(ForwardRef(InputBase))>
        </ForwardRef(Input)>
      </WithStyles(ForwardRef(Input))>
    </div>
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem",
        }
      }
      variant="contained"
    >
      <ForwardRef(Button)
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit",
            "contained": "MuiButton-contained",
            "containedPrimary": "MuiButton-containedPrimary",
            "containedSecondary": "MuiButton-containedSecondary",
            "containedSizeLarge": "MuiButton-containedSizeLarge",
            "containedSizeSmall": "MuiButton-containedSizeSmall",
            "disableElevation": "MuiButton-disableElevation",
            "disabled": "Mui-disabled",
            "endIcon": "MuiButton-endIcon",
            "focusVisible": "Mui-focusVisible",
            "fullWidth": "MuiButton-fullWidth",
            "iconSizeLarge": "MuiButton-iconSizeLarge",
            "iconSizeMedium": "MuiButton-iconSizeMedium",
            "iconSizeSmall": "MuiButton-iconSizeSmall",
            "label": "MuiButton-label",
            "outlined": "MuiButton-outlined",
            "outlinedPrimary": "MuiButton-outlinedPrimary",
            "outlinedSecondary": "MuiButton-outlinedSecondary",
            "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
            "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
            "root": "MuiButton-root",
            "sizeLarge": "MuiButton-sizeLarge",
            "sizeSmall": "MuiButton-sizeSmall",
            "startIcon": "MuiButton-startIcon",
            "text": "MuiButton-text",
            "textPrimary": "MuiButton-textPrimary",
            "textSecondary": "MuiButton-textSecondary",
            "textSizeLarge": "MuiButton-textSizeLarge",
            "textSizeSmall": "MuiButton-textSizeSmall",
          }
        }
        color="primary"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "margin": "1rem",
          }
        }
        variant="contained"
      >
        <WithStyles(ForwardRef(ButtonBase))
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled"
          component="button"
          disabled={true}
          focusRipple={true}
          focusVisibleClassName="Mui-focusVisible"
          onClick={[Function]}
          style={
            Object {
              "margin": "1rem",
            }
          }
          type="button"
        >
          <ForwardRef(ButtonBase)
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            component="button"
            disabled={true}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            onClick={[Function]}
            style={
              Object {
                "margin": "1rem",
              }
            }
            type="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "margin": "1rem",
                }
              }
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Get data
              </span>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(Button)>
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "1rem",
        }
      }
      variant="contained"
    >
      <ForwardRef(Button)
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit",
            "contained": "MuiButton-contained",
            "containedPrimary": "MuiButton-containedPrimary",
            "containedSecondary": "MuiButton-containedSecondary",
            "containedSizeLarge": "MuiButton-containedSizeLarge",
            "containedSizeSmall": "MuiButton-containedSizeSmall",
            "disableElevation": "MuiButton-disableElevation",
            "disabled": "Mui-disabled",
            "endIcon": "MuiButton-endIcon",
            "focusVisible": "Mui-focusVisible",
            "fullWidth": "MuiButton-fullWidth",
            "iconSizeLarge": "MuiButton-iconSizeLarge",
            "iconSizeMedium": "MuiButton-iconSizeMedium",
            "iconSizeSmall": "MuiButton-iconSizeSmall",
            "label": "MuiButton-label",
            "outlined": "MuiButton-outlined",
            "outlinedPrimary": "MuiButton-outlinedPrimary",
            "outlinedSecondary": "MuiButton-outlinedSecondary",
            "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
            "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
            "root": "MuiButton-root",
            "sizeLarge": "MuiButton-sizeLarge",
            "sizeSmall": "MuiButton-sizeSmall",
            "startIcon": "MuiButton-startIcon",
            "text": "MuiButton-text",
            "textPrimary": "MuiButton-textPrimary",
            "textSecondary": "MuiButton-textSecondary",
            "textSizeLarge": "MuiButton-textSizeLarge",
            "textSizeSmall": "MuiButton-textSizeSmall",
          }
        }
        color="primary"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "margin": "1rem",
          }
        }
        variant="contained"
      >
        <WithStyles(ForwardRef(ButtonBase))
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled"
          component="button"
          disabled={true}
          focusRipple={true}
          focusVisibleClassName="Mui-focusVisible"
          onClick={[Function]}
          style={
            Object {
              "margin": "1rem",
            }
          }
          type="button"
        >
          <ForwardRef(ButtonBase)
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            component="button"
            disabled={true}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            onClick={[Function]}
            style={
              Object {
                "margin": "1rem",
              }
            }
            type="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "margin": "1rem",
                }
              }
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Get favorites
              </span>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(Button)>
    </WithStyles(ForwardRef(Button))>
  </div>
</SearchBar>
`;
